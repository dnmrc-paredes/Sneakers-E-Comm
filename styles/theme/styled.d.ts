import 'styled-components'

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: {
      orange: string
      pale_orange: string
      light_gray: string
    }
    fonts: {
      nunito: string
      kumbh: string
    }
  }
}
