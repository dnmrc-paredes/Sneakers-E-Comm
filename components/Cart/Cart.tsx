import Image from 'next/image'
import styled from 'styled-components'
import { MdOutlineDelete } from 'react-icons/md'
import { useCart } from '../../contexts/CartContext'
import { breakpoints } from '../../styles'
import { useWindow } from '../../hooks'

const S = {
  Root: styled.div`
    position: absolute;
    width: 100%;
    top: 20px;
    z-index: 10;
    border-radius: 8px;
    right: 0;
    min-width: 340px;

    @media (min-width: ${breakpoints.md}) {
      left: -315px;
    }
  `,
  Wrapper: styled.div`
    border-radius: 8px;
    background-color: white;
    box-shadow: 2px 10px 34px -2px rgba(191, 191, 191, 0.75);

    @media (min-width: ${breakpoints.md}) {
      margin: 20px 10px;
    }
  `,
  TitleContainer: styled.div`
    flex: 1;
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    padding: 1rem;
    border-bottom: solid #c4c3c3 1px;

    h4 {
      margin: 0.5rem 1rem;
      font-family: ${({ theme }) => theme.fonts.kumbh};
    }

    button {
      background: none;
      border: none;
      cursor: pointer;
      font-weight: 700;
      width: 30px;
      height: 30px;
    }
  `,
  ItemsContainer: styled.div`
    flex: 2;
    display: flex;
    padding: 1rem 0;
    margin: 0.5rem 1rem;

    h4 {
      margin: auto;
      font-family: ${({ theme }) => theme.fonts.kumbh};
      color: ${({ theme }) => theme.colors.light_gray};
    }
  `,
  Item: styled.div`
    display: flex;

    img {
      border-radius: 5px;
    }
  `,
  ImageContainer: styled.div`
    flex: 1;
  `,
  Info: styled.div`
    flex: 2;
    margin-left: 10px;
    display: flex;
    justify-content: center;
    .info {
      flex: 1;
      display: flex;
      justify-content: center;
      flex-direction: column;

      h4,
      p {
        font-family: ${({ theme }) => theme.fonts.kumbh};
        font-weight: 400;

        span {
          font-weight: 800;
        }
      }
    }
    .delete {
      display: flex;
      justify-content: center;
      align-items: center;
      margin-left: 10px;
    }
  `,
  Button: styled.button`
    background-color: ${({ theme }) => theme.colors.orange};
    height: 50px;
    width: 90%;
    margin: 0.5rem 1rem;
    border: none;
    outline: none;
    font-family: ${({ theme }) => theme.fonts.kumbh};
    font-weight: 700;
    letter-spacing: 1px;
    color: white;
    cursor: pointer;
    border-radius: 10px;
    margin-bottom: 15px;
  `
}

export const Cart = ({
  handleToggleCart
}: {
  handleToggleCart: () => void
}) => {
  const { cartItems, setCartItems } = useCart()
  const { width } = useWindow()
  const isSmall = (width as number) < 1279

  const handleEmptyCart = () => {
    setCartItems(0)
  }

  return (
    <S.Root>
      <S.Wrapper>
        <S.TitleContainer>
          <h4> Cart </h4>
          {isSmall && (
            <button onClick={handleToggleCart} type='button'>
              X
            </button>
          )}
        </S.TitleContainer>
        <S.ItemsContainer>
          {cartItems ? (
            <S.Item data-testid='product'>
              <Image
                src='/image-product-1-thumbnail.jpg'
                width={50}
                height={50}
                alt='product'
              />
              <S.Info>
                <div className='info'>
                  <h4> Fall Limited Edition Sneakers </h4>
                  <p>
                    $125.00 x {cartItems} <span> ${125.0 * cartItems} </span>
                  </p>
                </div>
                <div className='delete'>
                  <MdOutlineDelete
                    onClick={handleEmptyCart}
                    color='grey'
                    size={20}
                    style={{ cursor: 'pointer' }}
                  />
                </div>
              </S.Info>
            </S.Item>
          ) : (
            <h4 data-testid='empty-cart'> Your cart is empty </h4>
          )}
        </S.ItemsContainer>
        <S.Button>Checkout</S.Button>
      </S.Wrapper>
    </S.Root>
  )
}
