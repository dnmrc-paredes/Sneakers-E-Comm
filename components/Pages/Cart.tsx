import React, { Dispatch, SetStateAction } from 'react'
import styled from 'styled-components'
import Image from 'next/image'
import { MdOutlineDelete } from 'react-icons/md'

const Base = {
  Root: styled.div`
    position: absolute;
    width: 340px;
    top: 50px;
    right: -120px;
    box-shadow: 0px 1px 120px 1px #c4c3c3;
    z-index: 10;
    border-radius: 8px;
    @media screen and (min-width: 300px) and (max-width: 550px) {
      position: fixed;
      top: 60px;
      left: 0;
      right: 0;
      bottom: 0;
      transform: none;
      width: 100%;
      height: max-content;
      box-shadow: none;
      /* width: max-content; */
    }
    @media screen and (min-width: 551px) and (max-width: 768px) {
      right: -20px;
    }
    @media screen and (min-width: 769px) and (max-width: 1000px) {
      right: -60px;
    }
  `,
  SubRoot: styled.div`
    border-radius: 8px;
    background-color: white;
    @media screen and (min-width: 300px) and (max-width: 550px) {
      margin: 20px 10px;
    }
  `,
  TitleContainer: styled.div`
    flex: 1;
    display: flex;
    align-items: center;
    padding: 1rem 0;
    border-bottom: solid #c4c3c3 1px;

    h4 {
      margin: 0.5rem 1rem;
      font-family: ${({ theme }) => theme.fonts.kumbh};
    }
  `,
  ItemsContainer: styled.div`
    flex: 2;
    display: flex;
    padding: 1rem 0;
    margin: 0.5rem 1rem;

    h4 {
      margin: auto;
      font-family: ${({ theme }) => theme.fonts.kumbh};
      color: ${({ theme }) => theme.colors.light_gray};
    }
  `,
  Item: styled.div`
    display: flex;

    img {
      border-radius: 5px;
    }
  `,
  ImageContainer: styled.div`
    flex: 1;
  `,
  Info: styled.div`
    flex: 2;
    margin-left: 10px;
    display: flex;
    justify-content: center;
    .info {
      flex: 1;
      display: flex;
      justify-content: center;
      flex-direction: column;

      h4,
      p {
        font-family: ${({ theme }) => theme.fonts.kumbh};
        font-weight: 400;

        span {
          font-weight: 800;
        }
      }
    }
    .delete {
      display: flex;
      justify-content: center;
      align-items: center;
      margin-left: 10px;
    }
  `,
  Button: styled.button`
    background-color: ${({ theme }) => theme.colors.orange};
    height: 50px;
    width: 90%;
    margin: 0.5rem 1rem;
    border: none;
    outline: none;
    font-family: ${({ theme }) => theme.fonts.kumbh};
    font-weight: 700;
    letter-spacing: 1px;
    color: white;
    cursor: pointer;
    border-radius: 10px;
    margin-bottom: 15px;
  `
}

type CartProps = {
  cartItems: number
  setCartItems: Dispatch<SetStateAction<number>>
}

export const Cart = ({ cartItems, setCartItems }: CartProps) => {
  return (
    <Base.Root>
      <Base.SubRoot>
        <Base.TitleContainer>
          <h4> Cart </h4>
        </Base.TitleContainer>
        <Base.ItemsContainer>
          {cartItems ? (
            <Base.Item>
              <Image
                src='/image-product-1-thumbnail.jpg'
                width={50}
                height={50}
                alt='product'
              />
              <Base.Info>
                <div className='info'>
                  <h4> Fall Limited Edition Sneakers </h4>
                  <p>
                    $125.00 x {cartItems} <span> ${125.0 * cartItems} </span>
                  </p>
                </div>
                <div className='delete'>
                  <MdOutlineDelete
                    onClick={() => setCartItems(0)}
                    color='grey'
                    size={20}
                    style={{ cursor: 'pointer' }}
                  />
                </div>
              </Base.Info>
            </Base.Item>
          ) : (
            <h4> Your cart is empty </h4>
          )}
        </Base.ItemsContainer>
        <Base.Button>Checkout</Base.Button>
      </Base.SubRoot>
    </Base.Root>
  )
}
