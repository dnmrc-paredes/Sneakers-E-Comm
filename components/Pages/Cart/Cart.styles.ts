import styled from 'styled-components'

export const Root = styled.div`
  position: absolute;
  width: 340px;
  top: 50px;
  right: -120px;
  box-shadow: 2px 10px 34px -2px rgba(191, 191, 191, 0.75);
  z-index: 10;
  border-radius: 8px;
  @media screen and (min-width: 300px) and (max-width: 550px) {
    position: fixed;
    top: 60px;
    left: 0;
    right: 0;
    bottom: 0;
    transform: none;
    width: 100%;
    height: max-content;
    box-shadow: none;
  }
  @media screen and (min-width: 551px) and (max-width: 768px) {
    right: -20px;
  }
  @media screen and (min-width: 769px) and (max-width: 1000px) {
    right: -60px;
  }
`

export const SubRoot = styled.div`
  border-radius: 8px;
  background-color: white;
  @media screen and (min-width: 300px) and (max-width: 550px) {
    margin: 20px 10px;
    background-color: white;
    box-shadow: 2px 10px 34px -2px rgba(97, 97, 97, 0.75);
  }
`

export const TitleContainer = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
  padding: 1rem 0;
  border-bottom: solid #c4c3c3 1px;

  h4 {
    margin: 0.5rem 1rem;
    font-family: ${({ theme }) => theme.fonts.kumbh};
  }
`

export const ItemsContainer = styled.div`
  flex: 2;
  display: flex;
  padding: 1rem 0;
  margin: 0.5rem 1rem;

  h4 {
    margin: auto;
    font-family: ${({ theme }) => theme.fonts.kumbh};
    color: ${({ theme }) => theme.colors.light_gray};
  }
`

export const Item = styled.div`
  display: flex;

  img {
    border-radius: 5px;
  }
`

export const ImageContainer = styled.div`
  flex: 1;
`

export const Info = styled.div`
  flex: 2;
  margin-left: 10px;
  display: flex;
  justify-content: center;
  .info {
    flex: 1;
    display: flex;
    justify-content: center;
    flex-direction: column;

    h4,
    p {
      font-family: ${({ theme }) => theme.fonts.kumbh};
      font-weight: 400;

      span {
        font-weight: 800;
      }
    }
  }
  .delete {
    display: flex;
    justify-content: center;
    align-items: center;
    margin-left: 10px;
  }
`

export const Button = styled.button`
  background-color: ${({ theme }) => theme.colors.orange};
  height: 50px;
  width: 90%;
  margin: 0.5rem 1rem;
  border: none;
  outline: none;
  font-family: ${({ theme }) => theme.fonts.kumbh};
  font-weight: 700;
  letter-spacing: 1px;
  color: white;
  cursor: pointer;
  border-radius: 10px;
  margin-bottom: 15px;
`
