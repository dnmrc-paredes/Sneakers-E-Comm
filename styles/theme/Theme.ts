export const Theme = {
  colors: {
    orange: 'hsl(26, 100%, 55%)',
    pale_orange: 'hsl(25, 100%, 94%)',
    light_gray: 'hsl(220, 14%, 75%)'
  },
  fonts: {
    nunito: 'Nunito Sans',
    kumbh: 'Kumbh Sans'
  }
}
