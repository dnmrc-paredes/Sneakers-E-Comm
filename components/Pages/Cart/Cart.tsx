import React, { Dispatch, SetStateAction } from 'react'
import Image from 'next/image'
import * as Base from './Cart.styles'
import { MdOutlineDelete } from 'react-icons/md'

type CartProps = {
  cartItems: number
  setCartItems: Dispatch<SetStateAction<number>>
}

export const Cart = ({ cartItems, setCartItems }: CartProps) => {
  return (
    <Base.Root>
      <Base.SubRoot>
        <Base.TitleContainer>
          <h4> Cart </h4>
        </Base.TitleContainer>
        <Base.ItemsContainer>
          {cartItems ? (
            <Base.Item data-testid="product">
              <Image
                src='/image-product-1-thumbnail.jpg'
                width={50}
                height={50}
                alt='product'
              />
              <Base.Info>
                <div className='info'>
                  <h4> Fall Limited Edition Sneakers </h4>
                  <p>
                    $125.00 x {cartItems} <span> ${125.0 * cartItems} </span>
                  </p>
                </div>
                <div className='delete'>
                  <MdOutlineDelete
                    onClick={() => setCartItems(0)}
                    color='grey'
                    size={20}
                    style={{ cursor: 'pointer' }}
                  />
                </div>
              </Base.Info>
            </Base.Item>
          ) : (
            <h4 data-testid="empty-cart"> Your cart is empty </h4>
          )}
        </Base.ItemsContainer>
        <Base.Button>Checkout</Base.Button>
      </Base.SubRoot>
    </Base.Root>
  )
}
