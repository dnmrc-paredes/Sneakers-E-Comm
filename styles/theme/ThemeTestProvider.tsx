import React from 'react'
import { ThemeProvider } from 'styled-components'
import { Theme } from './Theme'

export const ThemeTestProvider = ({
  children
}: {
  children: React.ReactNode
}) => {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>
}
