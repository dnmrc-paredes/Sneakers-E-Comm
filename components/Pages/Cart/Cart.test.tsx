import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { ThemeTestProvider } from '../../../styles/theme/ThemeTestProvider'
import { Cart } from './Cart'

describe('Cart component', () => {
  let cartItems = 0
  const setCartItems = jest.fn(() => (cartItems += 1))

  it('Render the Cart with no items', () => {
    const { getByTestId } = render(
      <ThemeTestProvider>
        <Cart cartItems={cartItems} setCartItems={setCartItems} />
      </ThemeTestProvider>
    )
    expect(getByTestId('empty-cart')).toBeInTheDocument()
  })

  it('Render the Cart with item', () => {
    setCartItems()
    const { getByTestId } = render(
      <ThemeTestProvider>
        <Cart cartItems={cartItems} setCartItems={setCartItems} />
      </ThemeTestProvider>
    )
    expect(getByTestId('product')).toBeInTheDocument()
  })

  it('Match the snapshot', () => {
    const tree = renderer
      .create(
        <ThemeTestProvider>
          <Cart cartItems={cartItems} setCartItems={setCartItems} />
        </ThemeTestProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
