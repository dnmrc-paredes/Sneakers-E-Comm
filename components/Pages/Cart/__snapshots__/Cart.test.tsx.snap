// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component Match the snapshot 1`] = `
<div
  className="Cartstyles__Root-sc-1w7brp9-0 igDYJn"
>
  <div
    className="Cartstyles__SubRoot-sc-1w7brp9-1 iREviS"
  >
    <div
      className="Cartstyles__TitleContainer-sc-1w7brp9-2 akCMy"
    >
      <h4>
         Cart 
      </h4>
    </div>
    <div
      className="Cartstyles__ItemsContainer-sc-1w7brp9-3 gLwdwf"
    >
      <div
        className="Cartstyles__Item-sc-1w7brp9-4 ZigIW"
        data-testid="product"
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAiIGhlaWdodD0iNTAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="product"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="product"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/_next/image?url=%2Fimage-product-1-thumbnail.jpg&w=128&q=75"
              srcSet="/_next/image?url=%2Fimage-product-1-thumbnail.jpg&w=64&q=75 1x, /_next/image?url=%2Fimage-product-1-thumbnail.jpg&w=128&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
        <div
          className="Cartstyles__Info-sc-1w7brp9-6 lbpJcv"
        >
          <div
            className="info"
          >
            <h4>
               Fall Limited Edition Sneakers 
            </h4>
            <p>
              $125.00 x 
              1
               
              <span>
                 $
                125
                 
              </span>
            </p>
          </div>
          <div
            className="delete"
          >
            <svg
              color="grey"
              fill="currentColor"
              height={20}
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": "grey",
                  "cursor": "pointer",
                }
              }
              viewBox="0 0 24 24"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M16 9v10H8V9h8m-1.5-6h-5l-1 1H5v2h14V4h-3.5l-1-1zM18 7H6v12c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <button
      className="Cartstyles__Button-sc-1w7brp9-7 fLpqnN"
    >
      Checkout
    </button>
  </div>
</div>
`;
